src/warc2zim/content_rewriting/ds.py

```diff
@@ -0,0 +1,269 @@
+## THis file is almost a plain translation from js file
+## https://github.com/webrecorder/wabac.js/blob/219830cea0a732bdd72ce100fcfd2f3a1c9c3607/src/rewrite/dsruleset.js
+## Partly translated by a human, partly by chat-gpt.
+
+## The rules set should be keep in sync with the original ones, but there is no easy
+## automatical process to do for now.
+
+
+import json
+import re
+from dataclasses import dataclass
+
+from warc2zim.content_rewriting.js import (
+    JsRewriter,
+    TransformationAction,
+    TransformationRule,
+    add_prefix,
+    m2str,
+)
+from warc2zim.content_rewriting.rx_replacer import add_around, add_suffix, replace_all
+
+
+@dataclass
+class SpecificRules:
+    contains: list[str]
+    rx_rules: list[TransformationRule]
+
+
+MAX_BITRATE = 5000000
+
+
+def set_max_bitrate(_opts: dict) -> int:
+    max_bitrate = MAX_BITRATE
+
+    # Extra opts is used in wabac, but we never set it ourself here.
```

Comment:
I'm sorry but I don't get what you mean here. Where do these `extraOpts` come from? The Twitter/Vimeo response? Then why don't we process them just like wabac does?

Maybe you should simply open an issue explaining:
- why we do not set extra_opts (we never did?)
- what could be the driver to decide to set them

Reply:
We are two :)

Wabac's `set_max_bitrate` is searching/saving data in `extraOpts`. But I don't get from where it comes from and what to do with it.

Reply:
I've opened https://github.com/openzim/warc2zim/issues/161 to track the question (at least)